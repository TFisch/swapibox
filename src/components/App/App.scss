.App {
  text-align: center;
  
}

.container-wrap {
  height: 100em;
  margin-top: 5em;
}