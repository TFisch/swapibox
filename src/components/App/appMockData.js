export const appMockData = [
  {
    Homeworld: "Tatooine",  
    Population: "200000",
    Species: "Human",
    isFavoriteSelected: false,
    name: "Luke Skywalker"
  }
];

export const expectedAppMock = [{
  "birth_year": "19BBY",
  "created": "2014-12-09T13:50:51.644000Z",
  "edited": "2014-12-20T21:17:56.891000Z",
  "eye_color": "blue",
  "films": ["https://swapi.co/api/films/2/", "https://swapi.co/api/films/6/", "https://swapi.co/api/films/3/", "https://swapi.co/api/films/1/", "https://swapi.co/api/films/7/"],
  "gender": "male",
  "hair_color": "blond",
  "height": "172",
  "homeworld": "https://swapi.co/api/planets/1/",
  "mass": "77",
  "name": "Luke Skywalker",
  "skin_color": "fair",
  "species": ["https://swapi.co/api/species/1/"],
  "starships": ["https://swapi.co/api/starships/12/", "https://swapi.co/api/starships/22/"],
  "stats": {
    "Homeworld": undefined,
    "Population": undefined,
    "Species": undefined
  },
  "url": "https://swapi.co/api/people/1/",
  "vehicles": ["https://swapi.co/api/vehicles/14/", "https://swapi.co/api/vehicles/30/"]
}, {
  "birth_year": "112BBY",
  "created": "2014-12-10T15:10:51.357000Z",
  "edited": "2014-12-20T21:17:50.309000Z",
  "eye_color": "yellow",
  "films": ["https://swapi.co/api/films/2/", "https://swapi.co/api/films/5/", "https://swapi.co/api/films/4/", "https://swapi.co/api/films/6/", "https://swapi.co/api/films/3/", "https://swapi.co/api/films/1/"],
  "gender": "n/a",
  "hair_color": "n/a",
  "height": "167",
  "homeworld": "https://swapi.co/api/planets/1/",
  "mass": "75",
  "name": "C-3PO",
  "skin_color": "gold",
  "species": ["https://swapi.co/api/species/2/"],
  "starships": [],
  "stats": {
    "Homeworld": undefined,
    "Population": undefined,
    "Species": undefined
  },
  "url": "https://swapi.co/api/people/2/",
  "vehicles": []
}];

export const appMockPeople = {
  "count": 87,
  "next": "https://swapi.co/api/people/?page=2",
  "previous": null,
  "results": [{
    "name": "Luke Skywalker",
    "height": "172",
    "mass": "77",
    "hair_color": "blond",
    "skin_color": "fair",
    "eye_color": "blue",
    "birth_year": "19BBY",
    "gender": "male",
    "homeworld": "https://swapi.co/api/planets/1/",
    "films": [
      "https://swapi.co/api/films/2/",
      "https://swapi.co/api/films/6/",
      "https://swapi.co/api/films/3/",
      "https://swapi.co/api/films/1/",
      "https://swapi.co/api/films/7/"
    ],
    "species": [
      "https://swapi.co/api/species/1/"
    ],
    "vehicles": [
      "https://swapi.co/api/vehicles/14/",
      "https://swapi.co/api/vehicles/30/"
    ],
    "starships": [
      "https://swapi.co/api/starships/12/",
      "https://swapi.co/api/starships/22/"
    ],
    "created": "2014-12-09T13:50:51.644000Z",
    "edited": "2014-12-20T21:17:56.891000Z",
    "url": "https://swapi.co/api/people/1/"
  }
        
  ]};

export const appMockSpecies = {
  "name": "Human",
  "classification": "mammal",
  "designation": "sentient",
  "average_height": "180",
  "skin_colors": "caucasian, black, asian, hispanic",
  "hair_colors": "blonde, brown, black, red",
  "eye_colors": "brown, blue, green, hazel, grey, amber",
  "average_lifespan": "120",
  "homeworld": "https://swapi.co/api/planets/9/",
  "language": "Galactic Basic",
  "people": [
    "https://swapi.co/api/people/1/",
    "https://swapi.co/api/people/4/",
    "https://swapi.co/api/people/5/",
    "https://swapi.co/api/people/6/",
    "https://swapi.co/api/people/7/",
    "https://swapi.co/api/people/9/",
    "https://swapi.co/api/people/10/",
    "https://swapi.co/api/people/11/",
    "https://swapi.co/api/people/12/",
    "https://swapi.co/api/people/14/",
    "https://swapi.co/api/people/18/",
    "https://swapi.co/api/people/19/",
    "https://swapi.co/api/people/21/",
    "https://swapi.co/api/people/22/",
    "https://swapi.co/api/people/25/",
    "https://swapi.co/api/people/26/",
    "https://swapi.co/api/people/28/",
    "https://swapi.co/api/people/29/",
    "https://swapi.co/api/people/32/",
    "https://swapi.co/api/people/34/",
    "https://swapi.co/api/people/43/",
    "https://swapi.co/api/people/51/",
    "https://swapi.co/api/people/60/",
    "https://swapi.co/api/people/61/",
    "https://swapi.co/api/people/62/",
    "https://swapi.co/api/people/66/",
    "https://swapi.co/api/people/67/",
    "https://swapi.co/api/people/68/",
    "https://swapi.co/api/people/69/",
    "https://swapi.co/api/people/74/",
    "https://swapi.co/api/people/81/",
    "https://swapi.co/api/people/84/",
    "https://swapi.co/api/people/85/",
    "https://swapi.co/api/people/86/",
    "https://swapi.co/api/people/35/"
  ],
  "films": [
    "https://swapi.co/api/films/2/",
    "https://swapi.co/api/films/7/",
    "https://swapi.co/api/films/5/",
    "https://swapi.co/api/films/4/",
    "https://swapi.co/api/films/6/",
    "https://swapi.co/api/films/3/",
    "https://swapi.co/api/films/1/"
  ],
  "created": "2014-12-10T13:52:11.567000Z",
  "edited": "2015-04-17T06:59:55.850671Z",
  "url": "https://swapi.co/api/species/1/"
};

export const appMockHomeworld = {
  "name": "Tatooine",
  "rotation_period": "23",
  "orbital_period": "304",
  "diameter": "10465",
  "climate": "arid",
  "gravity": "1 standard",
  "terrain": "desert",
  "surface_water": "1",
  "population": "200000",
  "residents": [
    "https://swapi.co/api/people/1/",
    "https://swapi.co/api/people/2/",
    "https://swapi.co/api/people/4/",
    "https://swapi.co/api/people/6/",
    "https://swapi.co/api/people/7/",
    "https://swapi.co/api/people/8/",
    "https://swapi.co/api/people/9/",
    "https://swapi.co/api/people/11/",
    "https://swapi.co/api/people/43/",
    "https://swapi.co/api/people/62/"
  ],
  "films": [
    "https://swapi.co/api/films/5/",
    "https://swapi.co/api/films/4/",
    "https://swapi.co/api/films/6/",
    "https://swapi.co/api/films/3/",
    "https://swapi.co/api/films/1/"
  ],
  "created": "2014-12-09T13:50:49.641000Z",
  "edited": "2014-12-21T20:48:04.175778Z",
  "url": "https://swapi.co/api/planets/1/"
};

export const appMockPlanet = {
  "count": 61,
  "next": "https://swapi.co/api/planets/?page=2",
  "previous": null,
  "results": [{
    "name": "Alderaan",
    "rotation_period": "24",
    "orbital_period": "364",
    "diameter": "12500",
    "climate": "temperate",
    "gravity": "1 standard",
    "terrain": "grasslands, mountains",
    "surface_water": "40",
    "population": "2000000000",
    "residents": [
      "https://swapi.co/api/people/5/",
      "https://swapi.co/api/people/68/",
      "https://swapi.co/api/people/81/"
    ],
    "films": [
      "https://swapi.co/api/films/6/",
      "https://swapi.co/api/films/1/"
    ],
    "created": "2014-12-10T11:35:48.479000Z",
    "edited": "2014-12-20T20:58:18.420000Z",
    "url": "https://swapi.co/api/planets/2/"
  },
  {
    "name": "Yavin IV",
    "rotation_period": "24",
    "orbital_period": "4818",
    "diameter": "10200",
    "climate": "temperate, tropical",
    "gravity": "1 standard",
    "terrain": "jungle, rainforests",
    "surface_water": "8",
    "population": "1000",
    "residents": [],
    "films": [
      "https://swapi.co/api/films/1/"
    ],
    "created": "2014-12-10T11:37:19.144000Z",
    "edited": "2014-12-20T20:58:18.421000Z",
    "url": "https://swapi.co/api/planets/3/"
  },
  {
    "name": "Hoth",
    "rotation_period": "23",
    "orbital_period": "549",
    "diameter": "7200",
    "climate": "frozen",
    "gravity": "1.1 standard",
    "terrain": "tundra, ice caves, mountain ranges",
    "surface_water": "100",
    "population": "unknown",
    "residents": [],
    "films": [
      "https://swapi.co/api/films/2/"
    ],
    "created": "2014-12-10T11:39:13.934000Z",
    "edited": "2014-12-20T20:58:18.423000Z",
    "url": "https://swapi.co/api/planets/4/"
  },
  {
    "name": "Dagobah",
    "rotation_period": "23",
    "orbital_period": "341",
    "diameter": "8900",
    "climate": "murky",
    "gravity": "N/A",
    "terrain": "swamp, jungles",
    "surface_water": "8",
    "population": "unknown",
    "residents": [],
    "films": [
      "https://swapi.co/api/films/2/",
      "https://swapi.co/api/films/6/",
      "https://swapi.co/api/films/3/"
    ],
    "created": "2014-12-10T11:42:22.590000Z",
    "edited": "2014-12-20T20:58:18.425000Z",
    "url": "https://swapi.co/api/planets/5/"
  },
  {
    "name": "Bespin",
    "rotation_period": "12",
    "orbital_period": "5110",
    "diameter": "118000",
    "climate": "temperate",
    "gravity": "1.5 (surface), 1 standard (Cloud City)",
    "terrain": "gas giant",
    "surface_water": "0",
    "population": "6000000",
    "residents": [
      "https://swapi.co/api/people/26/"
    ],
    "films": [
      "https://swapi.co/api/films/2/"
    ],
    "created": "2014-12-10T11:43:55.240000Z",
    "edited": "2014-12-20T20:58:18.427000Z",
    "url": "https://swapi.co/api/planets/6/"
  }
  ]};

export const appMockResidents =
    {
      "name": "Leia Organa",
      "height": "150",
      "mass": "49",
      "hair_color": "brown",
      "skin_color": "light",
      "eye_color": "brown",
      "birth_year": "19BBY",
      "gender": "female",
      "homeworld": "https://swapi.co/api/planets/2/",
      "films": [
        "https://swapi.co/api/films/2/",
        "https://swapi.co/api/films/6/",
        "https://swapi.co/api/films/3/",
        "https://swapi.co/api/films/1/",
        "https://swapi.co/api/films/7/"
      ],
      "species": [
        "https://swapi.co/api/species/1/"
      ],
      "vehicles": [
        "https://swapi.co/api/vehicles/30/"
      ],
      "starships": [],
      "created": "2014-12-10T15:20:09.791000Z",
      "edited": "2014-12-20T21:17:50.315000Z",
      "url": "https://swapi.co/api/people/5/"
    };

export const appMockVehicle = 
  {
    "count": 39,
    "next": "https://swapi.co/api/vehicles/?page=2",
    "previous": null,
    "results": [{
      "name": "Sand Crawler",
      "model": "Digger Crawler",
      "manufacturer": "Corellia Mining Corporation",
      "cost_in_credits": "150000",
      "length": "36.8",
      "max_atmosphering_speed": "30",
      "crew": "46",
      "passengers": "30",
      "cargo_capacity": "50000",
      "consumables": "2 months",
      "vehicle_class": "wheeled",
      "pilots": [],
      "films": [
        "https://swapi.co/api/films/5/",
        "https://swapi.co/api/films/1/"
      ],
      "created": "2014-12-10T15:36:25.724000Z",
      "edited": "2014-12-22T18:21:15.523587Z",
      "url": "https://swapi.co/api/vehicles/4/"
    },
    {
      "name": "T-16 skyhopper",
      "model": "T-16 skyhopper",
      "manufacturer": "Incom Corporation",
      "cost_in_credits": "14500",
      "length": "10.4",
      "max_atmosphering_speed": "1200",
      "crew": "1",
      "passengers": "1",
      "cargo_capacity": "50",
      "consumables": "0",
      "vehicle_class": "repulsorcraft",
      "pilots": [],
      "films": [
        "https://swapi.co/api/films/1/"
      ],
      "created": "2014-12-10T16:01:52.434000Z",
      "edited": "2014-12-22T18:21:15.552614Z",
      "url": "https://swapi.co/api/vehicles/6/"
    },
    {
      "name": "X-34 landspeeder",
      "model": "X-34 landspeeder",
      "manufacturer": "SoroSuub Corporation",
      "cost_in_credits": "10550",
      "length": "3.4",
      "max_atmosphering_speed": "250",
      "crew": "1",
      "passengers": "1",
      "cargo_capacity": "5",
      "consumables": "unknown",
      "vehicle_class": "repulsorcraft",
      "pilots": [],
      "films": [
        "https://swapi.co/api/films/1/"
      ],
      "created": "2014-12-10T16:13:52.586000Z",
      "edited": "2014-12-22T18:21:15.583700Z",
      "url": "https://swapi.co/api/vehicles/7/"
    },
    {
      "name": "TIE/LN starfighter",
      "model": "Twin Ion Engine/Ln Starfighter",
      "manufacturer": "Sienar Fleet Systems",
      "cost_in_credits": "unknown",
      "length": "6.4",
      "max_atmosphering_speed": "1200",
      "crew": "1",
      "passengers": "0",
      "cargo_capacity": "65",
      "consumables": "2 days",
      "vehicle_class": "starfighter",
      "pilots": [],
      "films": [
        "https://swapi.co/api/films/2/",
        "https://swapi.co/api/films/3/",
        "https://swapi.co/api/films/1/"
      ],
      "created": "2014-12-10T16:33:52.860000Z",
      "edited": "2014-12-22T18:21:15.606149Z",
      "url": "https://swapi.co/api/vehicles/8/"
    }
    ]};

