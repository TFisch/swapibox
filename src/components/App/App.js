import React, { Component } from 'react';
import './App.css';
import { MockData } from '../../data/MockData';
import CleanData from '../Helper/Helper';
import Welcome from '../Welcome/Welcome.js'
import Nav from '../Nav/Nav';
import CardContainer from '../CardContainer/CardContainer';

class App extends Component {
  constructor() {
    super()
    this.state = {
      crawlingText: {},
      data: [],
      favorites: [],
      cleanData: new CleanData()
    }
  }

  componentDidMount = () => {
    this.fetchAPI()
    this.fetchNameData()
  }
  
  fetchAPI = () => {
    const randomNumber = Math.floor(Math.random() * 6 + 1)
    fetch(`https://swapi.co/api/films/${randomNumber}/`)
    .then(response => response.json())
    .then(data => this.playOpeningScrawl(data))
    .catch((error) => console.log(error.message))
  }

  playOpeningScrawl = (data) => {
    const crawlingText = this.state.cleanData.getOpeningScrawl(data)
    this.setState({ crawlingText })
  }

  fetchNameData = (category) => {
    fetch(`https://swapi.co/api/people/`)
    .then(response => response.json())
    .then(data => this.state.cleanData.cleanHomeworld(data.results))
    .then(peopleData => this.state.cleanData.cleanSpecies(peopleData))
    .then(species => this.setState({data: species}))
    .catch((error) => console.log(error.message))
  };


  selectFavorite = (name) => {
    const favorites = this.state.favorites;
    const isFavorite = favorites.includes(name);
    const foundCard = favorites.find(card => card.name === name);

    if (isFavorite) {
      foundCard.isFavoriteSelected = !foundCard.isFavoriteSelected
    }

  }

  deselectFavorite = (name) => {
    
  }

  render() {
    const { data, crawlingText } = this.state
    return (
      <div className="App">
        <div>
          <Nav fetchNameData={this.fetchNameData}/>
          <Welcome crawlingText={crawlingText} />
          {/* <CardContainer data={data}/> */}
        </div>
      </div>
    );
  }
}

export default App;
