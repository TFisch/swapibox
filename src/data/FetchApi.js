import App from '../components/App/App.js';
import { getOpeningScrawl, cleanPeopleData, cleanSpecies, cleanVehicles, cleanPlanetData } from '../components/Helper/Helper.js';

export const fetchScrawl = async () => {
  const randomNumber = Math.floor(Math.random() * 6 + 1);
  let resolvedScrawl;
    const response = await fetch(`https://swapi.co/api/films/${randomNumber}/`);
    const data = await response.json();
    return (resolvedScrawl = await getOpeningScrawl(data));
};

export const fetchPeopleData = async () => {
  let peopleData;
    const response = await fetch(`https://swapi.co/api/people/`)
    return peopleData = await response.json()
};

export const fetchHomeWorld = async planet => {
  let homeworld;
    const response = await fetch(planet);
    const data = await response.json();
    return (homeworld = await data.name);
};

export const fetchPopulation = async planet => {
  let population;
    const response = await fetch(planet);
    const data = await response.json();
    return (population = await data.population);
};

export const fetchSpecies = async peopleData => {
  let species;
    const response = await fetch(peopleData);
    const data = await response.json();
    return (species = await data.name);
};

export const fetchPlanetData = async () => {
  let planetData
    const response = await fetch('https://swapi.co/api/planets/')
    return planetData = await response.json()
};

export const fetchResidents = async (residentLinks) => {
  let fetchPlanetResidents = []
  await residentLinks.reduce(async (residentNames, resident) => {
    const response = await fetch(resident)
    const residentList = await response.json()
    const residentName = await residentList.name
    residentNames = residentName
    fetchPlanetResidents.push(residentName)
    return residentNames
  }, '')
  return fetchPlanetResidents
};

export const fetchVehicleData = async () => {
    const response = await fetch(`https://swapi.co/api/vehicles/`);
    const data = await response.json();
    const vehicleData = await cleanVehicles(data);
    return vehicleData;
};