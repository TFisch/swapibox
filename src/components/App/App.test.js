import React from 'react';
import App from './App';
import { shallow } from 'enzyme';
import { appMockData, appMockVehicle, appMockPeople, expectedAppMock } from './appMockData';
import { films } from '../../data/MockData';

jest.mock('../../data/FetchApi');

describe('APP', () => {
  let wrapper;
  let mockData;
  let mockRemoveFavorite;

  beforeEach(() => {
    wrapper = shallow(<App removeFavorite={mockRemoveFavorite} />);
    mockData = {};
    mockRemoveFavorite = jest.fn();
  });

  it('Should match snapshot', () => {

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('Should start state crawlingText empty object and data empty array', () => {
    expect(wrapper.state().peopleData.length).toEqual(0);
    expect(wrapper.state().crawlingText).toEqual({});
  });

  it('Should update state of crawlingText when playOpeningScrawl is invoked', async () => {
    await wrapper.instance().playOpeningScrawl();

    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.state().peopleData.length).toEqual(0);
    expect(wrapper.state().crawlingText).toEqual(films);
  });

  describe('componentDidMount', () => {
    it('Should set the state crawlingText with componentDidMount', async () => {
      await wrapper.update();

      expect(wrapper.state().crawlingText).toEqual(mockData);
    });
  });

  describe('getPeopleData', () => {
    it('Should update state when invoked', async () => {
      const expected = expectedAppMock;

      await wrapper.instance().getPeopleData();

      expect(wrapper.state().favoritesDisplaying).toEqual(false);
      expect(wrapper.state().peopleData).toEqual(expected);
    });

    it('Should invoke setDisplayData when invoked', async () => {
      const mockSetDisplayData = jest.fn();
      wrapper = shallow(
        <App
          setDisplayData={mockSetDisplayData}
          peopleData={appMockPeople}
        />
      );
      const spy = spyOn(wrapper.instance(), 'getPeopleData');
      wrapper.instance().forceUpdate();

      await wrapper.instance().getPeopleData();

      expect(spy).toHaveBeenCalled();
      expect(wrapper.state().favoritesDisplaying).toEqual(false);
    });

    it('Should invoke setDisplayData and setState when there is peopleData', async () => {
      const mockSetDisplayData = jest.fn();
      wrapper = shallow(
        <App setDisplayData={mockSetDisplayData} />);
      wrapper.setState({ peopleData: appMockData });
      const spy = spyOn(wrapper.instance(), 'getPeopleData');
      wrapper.instance().forceUpdate();

      await wrapper.instance().getPeopleData();

      expect(spy).toHaveBeenCalled();
      await expect(wrapper.state().favoritesDisplaying).toEqual(false);
    });
  });

  describe('getVehicleData', () => {
    it('Should set state when invoked', async () => {
      const mockSetDisplayData = jest.fn();
      wrapper = shallow(<App setDisplayData={mockSetDisplayData} />);
      wrapper.setState({ vehicleData: appMockVehicle });
      const expected = appMockVehicle;

      await wrapper.instance().getVehicleData();

      expect(wrapper.state().vehicleData).toEqual(expected);
      expect(wrapper.state().displayData).toEqual(appMockVehicle);
      expect(wrapper.state().favoritesDisplaying).toEqual(false);
    });

    it('Should invoke setDisplayData and setState when vehicleData', async () => {
      const mockSetDisplayData = jest.fn();
      wrapper = shallow(
        <App
          setDisplayData={mockSetDisplayData}
          vehicleData={appMockVehicle}
        />
      );
      const spy = spyOn(wrapper.instance(), 'getVehicleData');
      wrapper.instance().forceUpdate();

      await wrapper.instance().getVehicleData();

      expect(spy).toHaveBeenCalled();
      expect(wrapper.state().favoritesDisplaying).toEqual(false);
    });
  });

  describe('getPlanetData', () => {
    it('Should set the state planetData when invoked', async () => {
      await wrapper.instance().getPlanetData();

      expect(wrapper.state().planetData).toEqual([]);
      expect(wrapper.state().favoritesDisplaying).toEqual(false);
    });
  });

  describe('setDisplayData', () => {
    it('Should setState when invoked', () => {
      mockData = appMockData;
      wrapper.instance().setDisplayData(mockData);

      expect(wrapper.state().displayData.length).toEqual(1);
      expect(wrapper.state().displayData).toEqual(mockData);
    });
  });

  describe('selectFavorite', () => {
    let mockData;

    beforeEach(() => {
      mockData = appMockData;
    });

    it('Should update state when invoked', () => {
      expect(wrapper.state().favorites.length).toEqual(0);
      expect(wrapper.state().favoriteCount).toEqual(0);

      wrapper.setState({ displayData: mockData });
      wrapper.instance().selectFavorite("Luke Skywalker");

      expect(wrapper.state().favorites.length).toEqual(1);
      expect(wrapper.state().favoriteCount).toEqual(1);
    });
    it('Should update state when card is selected twice', () => {
      wrapper.setState({ displayData: mockData });
      wrapper.setState({ favoritesDisplaying: true });
      wrapper.instance().selectFavorite("Luke Skywalker");

      expect(wrapper.state().favorites.length).toEqual(1);
      expect(wrapper.state().favoriteCount).toEqual(1);

      wrapper.instance().selectFavorite("Luke Skywalker");

      expect(wrapper.state().favorites.length).toEqual(0);
      expect(wrapper.state().favoriteCount).toEqual(0);
    });
  });

  describe('toggleFavorites', () => {
    it('Should setState when invoked', () => {
      wrapper.instance().toggleFavorites();

      expect(wrapper.state().favoritesDisplaying).toEqual(true);
    });

    it('Should invoke setDisplayData when invoked', () => {
      const mockSetDisplayData = jest.fn();
      wrapper = shallow(<App setDisplayData={mockSetDisplayData} />);
      const spy = spyOn(wrapper.instance(), 'toggleFavorites');
      wrapper.instance().forceUpdate();
      wrapper.instance().toggleFavorites();

      expect(spy).toHaveBeenCalled();
    });
  });

  describe('removeFavorite', () => {
    it('Should invoke setDisplayData when invoked', () => {
      const mockSetDisplayData = jest.fn();
      wrapper = shallow(<App setDisplayData={mockSetDisplayData} />);
      const spy = spyOn(wrapper.instance(), 'removeFavorite');
      wrapper.instance().forceUpdate();
      wrapper.instance().removeFavorite();

      expect(spy).toHaveBeenCalled();
    });
  });
});