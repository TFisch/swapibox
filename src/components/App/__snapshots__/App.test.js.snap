// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`APP Should match snapshot 1`] = `"<div class=\\"App\\"><div class=\\"nav-wrapper\\"><div class=\\"header-row\\"><h1 class=\\"title-logo\\">SWAPI-Box</h1><span class=\\"fav-row\\"><button class=\\"undefined\\"></button><div class=\\"favorites\\">FAV PLACEHOLDER</div><div class=\\"fav-counter\\">0</div></span></div><div class=\\"button-row\\"><button class=\\"peopleBtn\\" value=\\"peopleBtn\\">peopleBtn</button><button class=\\"vehiclesBtn\\" value=\\"vehiclesBtn\\">vehiclesBtn</button><button class=\\"planetsBtn\\" value=\\"planetsBtn\\">planetsBtn</button></div></div><div class=\\"container-wrap\\"><div class=\\"card-container\\"><div class=\\"fade\\"></div><section class=\\"star-wars\\"><div class=\\"crawl\\"><div class=\\"title\\"><p>Episode </p><h1></h1></div><p></p></div></section></div></div></div>"`;

exports[`APP Should update state of crawlingText when playOpeningScrawl is invoked 1`] = `"<div class=\\"App\\"><div class=\\"nav-wrapper\\"><div class=\\"header-row\\"><h1 class=\\"title-logo\\">SWAPI-Box</h1><span class=\\"fav-row\\"><button class=\\"undefined\\"></button><div class=\\"favorites\\">FAV PLACEHOLDER</div><div class=\\"fav-counter\\">0</div></span></div><div class=\\"button-row\\"><button class=\\"peopleBtn\\" value=\\"peopleBtn\\">peopleBtn</button><button class=\\"vehiclesBtn\\" value=\\"vehiclesBtn\\">vehiclesBtn</button><button class=\\"planetsBtn\\" value=\\"planetsBtn\\">planetsBtn</button></div></div><div class=\\"container-wrap\\"><div class=\\"card-container\\"><div class=\\"fade\\"></div><section class=\\"star-wars\\"><div class=\\"crawl\\"><div class=\\"title\\"><p>Episode </p><h1></h1></div><p></p></div></section></div></div></div>"`;
